builtin_array.go

```diff
@@ -195,9 +195,15 @@ func (r *Runtime) arrayproto_join(call FunctionCall) Value {
 		buf.WriteString(sep)
 		element := o.self.getIdx(valueInt(int64(i)), nil)
 		if element != nil && element != _undefined && element != _null {
-			buf.WriteString(element.toString())
+			obj := element.ToObject(r)
+			if _, ok := r.seenObjects[obj]; ok {
```

Comment:
are we able to leverage the `SameAs` receiver on the Object struct?
ex: `obj.SameAs(o)`

Reply:
I originally was using that and it works for the case of `n.push(n)` but when you have a nested self-reference such as the case in `TestArrayCycleDouble`, the loop still occurs 😕